src/styles/main.scss

```diff
@@ -1,3 +1,144 @@
+$block-size: 100;
+$default-days: 31;
+
 body {
   margin: 0;
+  font-size: 30px;
+  font-family: Arial, serif;
+}
+
+.calendar {
+  margin-inline: auto;
+  display: flex;
+  padding: 20px 10px 0;
+
+  flex-wrap: wrap;
+  width: ($block-size * 7 + 10);
+  height: auto;
+  gap: 1px;
+
+  @for $day from 1 through $default-days {
+    & :nth-child(#{$day}) {
+      display: flex;
+
+      &::after {
+        content: "#{$day}";
+        content-visibility: visible;
+        position: absolute;
+      }
+    }
+  }
+
+  &--days--28 {
+    @for $day from 1 through $default-days {
+      & :nth-child(#{$day}) {
+        display: flex;
+
+        &::after {
+          content: "#{$day}";
+          content-visibility: visible;
+          position: absolute;
+        }
+      }
+
+      @if $day > 28 {
+        & :nth-child(#{$day}) {
+          display: none;
+        }
+      }
+    }
+  }
+
+  &--days--29 {
+    @for $day from 1 through $default-days {
+      & :nth-child(#{$day}) {
+        display: flex;
+
+        &::after {
+          content: "#{$day}";
+          content-visibility: visible;
+          position: absolute;
+        }
+      }
+
+      @if $day > 29 {
+        & :nth-child(#{$day}) {
+          display: none;
+        }
+      }
+    }
+  }
+
+  &--days--30 {
+    @for $day from 1 through $default-days {
+      & :nth-child(#{$day}) {
+        display: flex;
+
+        &::after {
+          content: "#{$day}";
+          content-visibility: visible;
+          position: absolute;
+        }
+      }
+
+      @if $day > 30 {
+        & :nth-child(#{$day}) {
+          display: none;
+        }
+      }
+    }
+  }
+
+  & :first-child {
+    margin-left: ($block-size * 6 + 6);
+  }
+
+  &--start-day {
+    &--monday :first-child {
+      margin-left: 0;
+    }
+
+    &--tuesday :first-child {
+      margin-left: ($block-size * 1 + 1);
+    }
```

Comment:
Need to remove it, and use the loop for day start modifier as on the task description